src/worker.cc

```diff
@@ -153,27 +158,42 @@ void Worker::newUnixSocketConnection(evconnlistener *listener, evutil_socket_t f
   }
 }
 
-Status Worker::listenTCP(const std::string &host, int port, int backlog) {
-  sockaddr_in sin{};
-  sin.sin_family = AF_INET;
-  evutil_inet_pton(AF_INET, host.data(), &(sin.sin_addr));
-  sin.sin_port = htons(port);
-  int fd = socket(AF_INET, SOCK_STREAM, 0);
-  int sock_opt = 1;
-  if (setsockopt(fd, SOL_SOCKET, SO_REUSEADDR, &sock_opt, sizeof(sock_opt)) < 0) {
-    return Status(Status::NotOK, evutil_socket_error_to_string(EVUTIL_SOCKET_ERROR()));
-  }
-  // to support multi-thread binding on macOS
-  if (setsockopt(fd, SOL_SOCKET, SO_REUSEPORT, &sock_opt, sizeof(sock_opt)) < 0) {
-    return Status(Status::NotOK, evutil_socket_error_to_string(EVUTIL_SOCKET_ERROR()));
+Status Worker::listenTCP(const std::string &host, int port, int af, int backlog) {
```

Comment:
i don't think `af` should be a argument. you can judge host inner to check ipv4 or ipv6. so when calling, we don't need two statements

Reply:
yes, i think it will be more generic.